### PR TITLE
Improving projects managing

### PR DESCRIPTION
**Pull request by kacper-cyra**
General changes:
- Changed data typing and mapping in API,
- Populating project data after updating project,
- Added access to cached data in onSuccess function

Components changes:
- Redirecting user to projects page after save after receiving successful response status,
- Updating projects cache after editing single project,
- Improved typing

### CODE REVIEW

File: frontend/src/api/ReferenceProject.api.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import api from './api.service'
 import * as sectionService from './Section.api'
-import { Section, ProjectData } from '../models'
+import { ProjectData } from '../models'
+import { Section, SectionData } from '../models/Section.model'
```
Comment (pdybowski) [inline frontend/src/api/ReferenceProject.api.ts:4]: you can import all of them from '../models' (see ProjectData)

File: frontend/src/pages/Admin/ManageReferenceProject/ManageReferenceProject.tsx [hunk 2]
```diff
@@ -52,17 +71,30 @@ const ManageReferenceProject = (props: any) => {
...
     setIsEdit(!isEdit)
   }
 
-  function handleInputChange(event: any) {
-    const target = event.target
+  function handleInputChange(event: React.ChangeEvent<HTMLInputElement>) {
+    const target = event.currentTarget
     const name = target.id
-    let value = target.value
+    let value: string | number = target.value
```
Comment (pdybowski) [inline frontend/src/pages/Admin/ManageReferenceProject/ManageReferenceProject.tsx:97]: is there any posibility to have a number from target.value?

**Review by pdybowski — COMMENTED** [review]
you can improve imports and typing

File: frontend/src/hooks/useQuery/useProject.ts [hunk 3]
```diff
@@ -6,18 +6,41 @@ import {
...
+                  data.sectionId.startDate,
+                ).toLocaleDateString(),
+                endDate: new Date(data.sectionId.endDate).toLocaleDateString(),
+              }
+            }
+            return project
+          })
+        }
+        return cachedData
+      })
```
Comment (KrystianKjjk) [inline frontend/src/hooks/useQuery/useProject.ts:42]: Invert this both "if" statement, it will look cleaner

**Pull request merged**